checks/shell_download_validate.go

```diff
@@ -518,6 +562,12 @@ func isUnpinnedPakageManagerDownload(node syntax.Node, cmd, pathfn string,
 		return true
 	}
 
+	// npm install
+	if isNpmUnpinnedDownload(c) {
+		dl.Warn("insecure (unpinned) download deteted in %v: '%v'", pathfn, cmd)
```

Comment:
we've updated these error message but the ones in the files slipped thru the crack.
replace with "insecure (not pinned by hash) download detected".
We need to use the new `Warn3` function:
```
Warn3(&checker.LogMessage{
						Path: pathfn, Type: checker.FileTypeSource, 
                                                   Offset: LINE_NUMBER, 
                                                   Snippet: cmd,
						Text: "insecure (not pinned by hash) download detected",
					})
```
If you're unsure how to get the line number, don't set it and add a comment `// TODO: set Offset`

Do you mind making this change `Warn` -> `Warn3` for the entire file? (just 4 occurences)